Ryujinx.Common/SystemInfo/LinuxSystemInfo.cs

```diff
@@ -1,19 +1,81 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
 using System.IO;
-using System.Linq;
 using System.Runtime.Versioning;
+using Ryujinx.Common.Logging;
 
 namespace Ryujinx.Common.SystemInfo
 {
     [SupportedOSPlatform("linux")]
-    internal class LinuxSystemInfo : SystemInfo
+    class LinuxSystemInfo : SystemInfo
     {
-        public override string CpuName { get; }
-        public override ulong RamSize { get; }
+        internal LinuxSystemInfo()
+        {
+            string cpuName = GetCpuidCpuName();
+
+            if (cpuName == null)
+            {
+                var cpuDict = new Dictionary<string, string>(StringComparer.Ordinal)
+                {
+                    ["model name"] = null,
+                    ["Processor"] = null,
+                    ["Hardware"] = null
+                };
+
+                ParseKeyValues("/proc/cpuinfo", cpuDict);
+
+                cpuName = cpuDict["model name"] ?? cpuDict["Processor"] ?? cpuDict["Hardware"] ?? "Unknown";
+            }
 
-        public LinuxSystemInfo()
+            var memDict = new Dictionary<string, string>(StringComparer.Ordinal)
+            {
+                ["MemTotal"] = null,
+                ["MemAvailable"] = null
+            };
+
+            ParseKeyValues("/proc/meminfo", memDict);
+
+            // Entries are in KB
+            _ = ulong.TryParse(memDict["MemTotal"]?.Split(' ')[0],
+                NumberStyles.Integer, CultureInfo.InvariantCulture, out ulong totalKB);
+            _ = ulong.TryParse(memDict["MemAvailable"]?.Split(' ')[0],
+                NumberStyles.Integer, CultureInfo.InvariantCulture, out ulong availableKB);
+
+            CpuName = $"{cpuName} ; {LogicalCoreCount} logical";
+            RamTotal = totalKB * 1024;
+            RamAvailable = availableKB * 1024;
+        }
+
+        private static void ParseKeyValues(string filePath, Dictionary<string, string> itemDict)
         {
-            CpuName = File.ReadAllLines("/proc/cpuinfo").Where(line => line.StartsWith("model name")).ToList()[0].Split(":")[1].Trim();
-            RamSize = ulong.Parse(File.ReadAllLines("/proc/meminfo")[0].Split(":")[1].Trim().Split(" ")[0]) * 1024;
+            if (!File.Exists(filePath))
+            {
+                Logger.Error?.Print(LogClass.Application, $"File '{filePath}' not found");
```

Comment:
```suggestion
                Logger.Error?.Print(LogClass.Application, $"File \"{filePath}\" not found");
```
We usually use quotes for that. I don't have a preference here, just making the suggestion for consistency.